src/libcore/macros.rs

```diff
@@ -542,7 +542,9 @@ macro_rules! unreachable {
 #[macro_export]
 #[stable(feature = "rust1", since = "1.0.0")]
 macro_rules! unimplemented {
-    () => (panic!("not yet implemented"))
+    () => (panic!("not yet implemented"));
+    ($msg:expr) => (unimplemented!("{}", $msg));
```

Comment:
Well if you only want to accept valid format strings, you could replace this line and the next one with `($($arg:tt)+) => (panic!("not yet implemented: {}", format_args!($($arg)+)));` which gives the same error message as other macros accepting format strings:

```
error: invalid reference to argument `0` (no arguments given)
 --> <anon>:7:20
  |
7 |     unimplemented!("foo {}")
  |                    ^^^^^^^^
```